remote/remote_state_s3.go

```diff
@@ -1135,7 +1135,7 @@ func checkIfSSEForS3Enabled(s3Client *s3.S3, config *RemoteStateConfigS3, terrag
 	for _, rule := range output.ServerSideEncryptionConfiguration.Rules {
 		if rule.ApplyServerSideEncryptionByDefault != nil {
 			if rule.ApplyServerSideEncryptionByDefault.SSEAlgorithm != nil {
-				if *rule.ApplyServerSideEncryptionByDefault.SSEAlgorithm == s3.ServerSideEncryptionAwsKms {
+				if *rule.ApplyServerSideEncryptionByDefault.SSEAlgorithm == config.BucketSSEAlgorithm {
```

Comment:
Instead of validating that the bucket has KMS encryption enabled, we want to validate that it matches the user's preference of encryption.